iofog-agent-daemon/src/main/java/org/eclipse/iofog/pruning/DockerPruningManager.java

```diff
@@ -91,6 +98,73 @@ public void start() throws Exception {
         }
     };
 
+    /**
+     * Gets list of unwanted docker images to be removed
+     * @return list
+     */
+    public Set<String> getUnwantedImagesList() {
+        Set<String> imageIDsToBePruned = new HashSet<>();
+        List<Image> images = docker.getImages();
+        LoggingService.logDebug(MODULE_NAME, "Total number of images already downloaded in the machine : " + images.size());
+        List<Container> nonIoFogContainers = docker.getRunningNonIofogContainers();
+        LoggingService.logDebug(MODULE_NAME, "Total number of running non iofog containers : " + nonIoFogContainers.size());
+
+        // Removes the non-ioFog running container from the images to be prune list
+        for (Container container : nonIoFogContainers) {
```

Comment:
Instead of nested loops, you could do this one-liner:
```
List<Image> ioFogImages = images.stream().filter(im -> nonIoFogContainers.stream().noneMatch(c -> c.getImageId().equals(im.getId()))).collect(Collectors.toList());
```
Same for next loop:
```
imageIDsToBePruned = ioFogImages.stream().filter(im -> microservices.stream().noneMatch(ms -> ms.getImageName().equals(im.getId()))).map(Image::getId).collect(Collectors.toSet());
```

Reply:
updated